### ISSUE TITLE
Lens constructs only work properly at toplevel

### ISSUE THREAD

**Issue by jamescheney** [links-lang/links#437]
Here's a small example:

```
var db = database "links";

var tracksTable =
	table "tracks"
	with (track: String, date: Int, rating: Int, album: String)
	tablekeys [["track", "album"]]
	from db;

fun foo() {
var tracksLens = lens tracksTable with { track -> date rating };
var tracks = lensget tracksLens;
tracks
}

foo()
```

which yields

```
*** Error: "Assert_failure core/sugartypes.ml:391:27"
```

It works if you remove the function declaration / call.  Weirdly, this version also works:
```
...
var foo = fun () {
var tracksLens = lens tracksTable with { track -> date rating };
var tracks = lensget tracksLens;
tracks
};

foo()
```
so perhaps the problem is a bug in how named function declarations are typechecked/desugared that is not present in the case for anonymous functions.

This seems to be because the "LensFunDepsLit" construct persists to `sugartypes.ml` but it should have been eliminated earlier.  The desugaring happens insde `type_check` in `typeSugar.ml`.

Observed in stock Links 0.8.

**Comment by jamescheney** [comment]
Has anyone (e.g. @rudihorn) looked at this any further?  Was it fixed by #454?

**Comment by jamescheney** [comment]
Looks like it is fixed either by #434 or #627.

**Comment by jamescheney** [comment]
Fixed by #627
